index.js

```diff
@@ -627,15 +643,22 @@ function flushEmitQueue(self) {
 
     if (item.err) {
       // emit the delayed error
-      item.ee.emit('error', item.err);
+      if (item.ee && item.ee.listenerCount('error') > 0) {
+        try {
+          item.ee.emit('error', item.err);
+        }
+        catch(emit_error){
+          self.logger.warn(`Multiparty failed to emit an 'error' event in 'flushEmitQueue' [cause = ${error_message(emit_error)}]`);
+        }
+      }
     }
   }
 }
 
 function handlePart(self, partStream) {
   beginFlush(self);
   var emitAndReleaseHold = holdEmitQueue(self, partStream);
-  partStream.on('end', function() {
+  partStream.on('done', function() {
```

Comment:
I notice in handle_error that it is removing the 'end' listener, and I wonder if it should also be removing the 'done' listener somewhere.

Reply:
I will remove the listener

Reply:
The changes are complete.  The 'done' listener is removed after handling the event and after an error.